flowmachine/flowmachine/core/query.py

```diff
@@ -163,7 +170,7 @@ def __format__(self, fmt=""):
         return f"<{', '.join(all_descriptions)}>"
 
     def __iter__(self):
-        con = self.connection.engine
+        con = get_db().engine
```

Comment:
One thing I'm wrestling a little with across this whole PR is where one should use the context var and where one should pass in the object you get from it.